pkg/globalnet/controllers/ipam/gatewaymonitor_test.go

```diff
@@ -104,74 +111,126 @@ var _ = Describe("Service monitoring", func() {
-			t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain,
-				ContainSubstring(service.Annotations[ipam.SubmarinerIpamGlobalIp]))
+		When("a Service Export is created before service", func() {
+			It("should assign a global IP and add an appropriate IP tables rule after service creation", func() {
+				t.createSvcExport(service)
+				t.ipt.AwaitNoRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIPSubstr))
 
-			service.Annotations[ipam.SubmarinerIpamGlobalIp] = "169.254.1.20"
-			_, err := t.k8sClient.CoreV1().Services(namespace).Update(service)
-			Expect(err).To(Succeed())
-			t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain,
-				ContainSubstring(service.Annotations[ipam.SubmarinerIpamGlobalIp]))
+				t.createService(service)
+				globalIP := t.awaitServiceGlobalIP(service.Name)
+				t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIP))
+			})
 		})
 	})
 
-	When("a Service with a global IP is removed", func() {
-		It("should remove the appropriate IP tables rule", func() {
-			globalIP := t.awaitServiceGlobalIP(service.Name, localCIDR)
-			t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIP))
+	Context("", func() {
+		JustBeforeEach(func() {
+			service = t.createService(service)
+			t.createSvcExport(service)
+			t.createEndpoint(newEndpointSpec(clusterID, t.hostName, localCIDR))
+		})
 
-			Expect(t.k8sClient.CoreV1().Services(namespace).Delete(service.Name, nil)).To(Succeed())
-			t.ipt.AwaitNoRule("nat", constants.SmGlobalnetIngressChain, ContainSubstring(globalIP))
+		When("a Service without a global IP is created", func() {
+			It("should assign a global IP and add an appropriate IP tables rule", func() {
+				t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain,
+					ContainSubstring(t.awaitServiceGlobalIP(service.Name)))
+			})
 		})
-	})
 
-	When("a Service in an excluded namespace is created", func() {
-		BeforeEach(func() {
-			service.Namespace = excludedNS
-			t.spec.ExcludeNS = []string{excludedNS}
+		When("a Service's global IP is updated", func() {
+			BeforeEach(func() {
+				service.Annotations[ipam.SubmarinerIpamGlobalIp] = "169.254.1.10"
+			})
+
+			It("should update the appropriate IP tables rule", func() {
+				t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain,
+					ContainSubstring(service.Annotations[ipam.SubmarinerIpamGlobalIp]))
+
+				service.Annotations[ipam.SubmarinerIpamGlobalIp] = "169.254.1.20"
+				_, err := t.k8sClient.CoreV1().Services(namespace).Update(service)
+				Expect(err).To(Succeed())
+				t.ipt.AwaitRule("nat", constants.SmGlobalnetIngressChain,
+					ContainSubstring(service.Annotations[ipam.SubmarinerIpamGlobalIp]))
+			})
 		})
 
-		It("should not assign a global IP", func() {
-			t.awaitNoServiceGlobalIP(service)
+		When("the Service Export is deleted", func() {
+			It("should eventually remove global IP and the IP tables rule", func() {
```

Comment:
I know this is a nit but with my suggested wording below, it implies that the service was previously exported.

```suggestion
		When("a Service with a global IP is unexported", func() {
                     It("should unassign its global IP and remove the appropriate IP tables rule", func() {
```